File: web/src/core/feature/exchange/domain/caseuse/GetCachedTodayExchangeCaseUse.ts
import type { CupExchange } from "../entity/CupExchange";
import type { ExchangeRepository } from "../repository/ExchangeRepository";

export class GetCachedTodayExchangeCaseUse {
    constructor(private readonly repository: ExchangeRepository) {}

    async execute(): Promise<CupExchange | null> {
        return this.repository.getCachedToday();
    }
}

File: app/src/main/java/com/elitec/alejotaller/feature/exchange/domain/caseUse/GetCachedTodayExchangeCaseUse.kt
package com.elitec.alejotaller.feature.exchange.domain.caseUse

import com.elitec.alejotaller.feature.exchange.domain.entity.CupExchange
import com.elitec.alejotaller.feature.exchange.domain.repository.ExchangeRepository

class GetCachedTodayExchangeCaseUse(
    private val repository: ExchangeRepository
) {
    suspend operator fun invoke(): CupExchange? = repository.getCachedToday()
}
